core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1230,6 +1230,27 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertEquals(500, consumer0.committed(tp2).offset)
   }
 
+  @Test
+  def testCommitInvalidOffsets() {
+      this.consumers.head.assign(List(tp).asJava)
+      val negativeOffset = Map((tp, new OffsetAndMetadata(-1))).asJava
+      val callback = new CountConsumerCommitCallback
+
+      // sync commit - negative offset
+      try {
+          this.consumers.head.commitSync(negativeOffset)
+          fail("Should have thrown IllegalArgumentException")
+      } catch {
+          case iae: IllegalArgumentException => // expected
+      }
+
+      // async commit - negative offset
+      this.consumers.head.commitAsync(negativeOffset, callback)
+      awaitCommitCallback(this.consumers.head, callback, timeoutMs=5000)
```

Comment:
It seems that the default `successCount` is 1 here so we are always going to wait for 5 seconds?